### PR TITLE
refactor data providers into standalone module

### PR DESCRIPTION
**Pull request by sebady**
This pull request refactors our Data Provider functionality into a standalone module that allows clients to leverage our DataProviders with testng automation.

### CODE REVIEW

**Comment by mach6** [comment]
@sebady I'm making changes and moving this to a branch.

**Comment by mach6** [comment]
closing and re-opening against new branch.

File: client/src/test/resources/AssociativeArrayOfLists.yaml [hunk 1]
```diff
@@ -1,6 +0,0 @@
-# Block associated array of lists (showing examples of both inline and blocked lists)
```
Comment (mach6) [inline client/src/test/resources/AssociativeArrayOfLists.yaml:1]: @sebady Is this an inadvertent removal? I see test skips in the new dataprovider module as a result of this file (and the other 6 below) removed as test resources

Comment (sebady) [inline client/src/test/resources/AssociativeArrayOfLists.yaml:1]: Yes this removal was accidental.  The files should have been moved into dataprovider module.

**Pull request closed without merging**